sympy/core/traversal.py

```diff
@@ -29,8 +29,8 @@ def iterargs(expr):
         yield i
         try:
             args.extend(i.args)
-        except TypeError:
-            pass  # for cases like f being an arg
+        except AttributeError:
```

Comment:
Do you have a test that can test this?

Reply:
Thanks for the review @smichr !

Actually - couldn't find an example beside the one I introduced in the next commit (where args of permutation is its array form with numbers).
This made me realize I better stick to the way other children of Basic are solving this problem, by reimplementing __getnewargs__ (like Numbers, Str, etc.)
So, I changed the commit above, and now this change is not needed for this pull-request anymore.
(I added it so CI would pass)

The options I see:
1. Keep this change without testing it (the original purpose here was AttributeError, right?)
2. Discard this change from this pull-request
3. Remove this error handling completely (seems to me it is not needed, although I'm not sure).
4. Add a test that implements a custom child of Basic with non-Basic args that would make this error handling useful.

Please let me know what you want me to do.